### PR TITLE
Fix to remove hard coded URLs and add https

### PR DESCRIPTION
**Pull request by pieterlukasse**
# What? Why?
This removes hard coded references to www.cbioportal.org and adds https to the external service calls that don't go via the internal proxy (this is important for cbioportal installations hosted on a https address).

# Notify reviewers
@adamabeshouse @alisman

### CODE REVIEW

File: src/pages/datasetView/DatasetList.tsx [hunk 1]
```diff
@@ -49,7 +49,7 @@ class CancerStudyCell extends React.Component<ICancerStudyCellProps,{}> {
         return (
             <span>
                 <a
-                    href={`http://www.cbioportal.org/study?id=${this.props.studyId}#summary`}
```
Comment (pieterlukasse) [inline src/pages/datasetView/DatasetList.tsx]: @inodb @alisman: it seems this hard coded reference to www.cbioportal.org has been there for quite some time. At the same time, the datasets page seems to work correctly on a local installation. How is this possible? Is there a global search+replace going on during build time?

File: src/shared/components/mutationTable/column/CosmicColumnFormatter.tsx [hunk 2]
```diff
@@ -95,7 +95,7 @@ export default class CosmicColumnFormatter
     {
         const cosmic:CosmicMutation[]|null = CosmicColumnFormatter.getData(data, cosmicData);
 
-        let value:number = -1;
+        let value:number|undefined = -1;
```
Comment (pieterlukasse) [inline src/shared/components/mutationTable/column/CosmicColumnFormatter.tsx]: another question is this compilation error fix: isn't the whole project compiled at each commit? How could a compilation error creep in?

File: src/pages/datasetView/DatasetList.tsx (hunk 1 above)
Comment (alisman) [reply src/pages/datasetView/DatasetList.tsx]: look like you had an old version of this file.  In the below, you can see this was fixed.  On the other hand, there are other references to our datahub.  Not sure if those should also be configurable?

https://github.com/cBioPortal/cbioportal-frontend/blob/master/src/pages/datasetView/DatasetList.tsx

File: src/shared/components/mutationTable/column/CosmicColumnFormatter.tsx (hunk 2 above)
Comment (alisman) [reply src/shared/components/mutationTable/column/CosmicColumnFormatter.tsx]: hmm.  you should not need to allow undefined here as far as i can see.  what is compiler error?

File: src/pages/datasetView/DatasetList.tsx (hunk 1 above)
Comment (pieterlukasse) [reply src/pages/datasetView/DatasetList.tsx]: indeed old code. Sorry for alarming you. Will make a new PR.

File: src/shared/components/mutationTable/column/CosmicColumnFormatter.tsx (hunk 2 above)
Comment (pieterlukasse) [reply src/shared/components/mutationTable/column/CosmicColumnFormatter.tsx]: was old code....didn't encounter this in new PR #664

**Pull request closed without merging**